sherpa/astro/io/crates_backend.py

```diff
@@ -285,121 +286,152 @@ def _require_tbl_col(crate, colname, cnames, make_copy=False,
 
-    col = crate.get_column(colname)
+    try:
+        return _require_col_list(crate, colname,
+                                 make_copy=make_copy,
+                                 fix_type=fix_type)
+    except IOErr:
+        return np.full(num, None)
+
+
+def _require_col_list(crate: TABLECrate,
+                      colname: str,
+                      make_copy: bool = False,
+                      fix_type: bool = False) -> np.ndarray:
+    """
+    check for new crates
+    """
+
+    try:
+        col = crate.get_column(colname)
+    except ValueError:
+        raise IOErr("reqcol", colname, crate.get_filename()) from None
+
     if col.is_varlen():
         values = col.get_fixed_length_array()
     else:
         values = col.values
 
     if make_copy:
         # Make a copy if a filename passed in
-        col = numpy.array(values)
+        col = np.array(values)
     else:
         # Use a reference if a crate passed in
-        col = numpy.asarray(values)
+        col = np.asarray(values)
 
     if fix_type:
-        col = col.astype(dtype)
+        col = col.astype(SherpaFloat)
 
     return col
 
 
-def _require_col_list(crate, colname, num, make_copy=False, fix_type=False,
-                      dtype=SherpaFloat):
-    """
-    check for new crates
-    """
-
-    # _try_col_list returns [None] * num if the column doesn't exist
-    # which is awkward to check for so use an explicit check.
-    #
-    if not crate.column_exists(colname):
-        raise IOErr('badcol', colname)
-    return _try_col_list(crate, colname, num, make_copy, fix_type, dtype)
-
-
-def _require_col(crate, colname, make_copy=False, fix_type=False,
-                 dtype=SherpaFloat):
-    """
-    checked for new crates
-    """
-    data = _try_col(crate, colname, make_copy, fix_type, dtype)
-    if data is None:
-        raise IOErr('badcol', colname)
-    return data
+def _require_col(crate: TABLECrate,
+                 colname: str,
+                 make_copy: bool = False,
+                 fix_type: bool = False,
+                 label: Optional[str] = None,
+                 dtype: type = SherpaFloat) -> np.ndarray:
+    """Return the column data or error out."""
 
+    try:
+        col = crate.get_column(colname)
+    except ValueError:
+        msg = colname if label is None else label
+        raise IOErr("reqcol", msg, crate.get_filename()) from None
 
-def _try_image(crate, make_copy=False, fix_type=False, dtype=SherpaFloat):
-    """
-    checked for new crates
-    """
+    if col.is_varlen():
+        values = col.get_fixed_length_array()
+    else:
+        values = col.values
 
     if make_copy:
         # Make a copy if a filename passed in
-        dat = pycrates.copy_piximgvals(crate).squeeze()
+        data = np.array(values).ravel()
+
     else:
         # Use a reference if a crate passed in
-        dat = pycrates.get_piximgvals(crate).squeeze()
+        data = np.asarray(values).ravel()
 
```

Comment:
```suggestion
    data = np.array(values, copy=make_copy).ravel()
```

Reply:
I really don't like the existing `make_copy` behaviour (we use it less in the pyfits backend). I believe I know where the concern came from, but I wonder if we now still need it. However, if we can just do it with this one line it's less of a concern.